src/me.rs

```diff
@@ -175,11 +173,15 @@ mod nasm {
     {
       if mem::size_of::<T>() == 2 && is_x86_feature_detected!("ssse3") && blk_h >= 4 && blk_w >= 4 {
         return unsafe {
+          let plane_org = &*(plane_org as *const _ as *const PlaneSlice<'_, u16>);
+          let plane_ref = &*(plane_ref as *const _ as *const PlaneSlice<'_, u16>);
```

Comment:
Wow, pointer casting is ugly. I guess that is by design so that we feel bad doing it. 😅